### PR TITLE
Underlines links created via markup

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

Render markdown link texts with an underline.

Before:
Screen Shot 2023-12-16 at 11 28 44 AM

After:
Screen Shot 2023-12-16 at 11 28 07 AM

## Technical Summary

[Jira Ticket
](https://dimagi-dev.atlassian.net/browse/USH-3932?focusedCommentId=302587&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-302587)

## Feature Flag

No feature flag

### Automated test coverage

tests exist for basic markdown rendering and updated for underlined link text

### QA Plan

no QA

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         ko.bindingHandlers.renderMarkdown = {
             update: function (element, valueAccessor) {
                 var value = ko.unwrap(valueAccessor());
-                value = markdown.render(value || '');
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js:13]: Are you sure unwrap cannot return null or undefined?

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js [hunk 2]
```diff
@@ -835,7 +835,7 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
             },
             help: {
                 update: function (options) {
-                    return options.data ? markdown.render(DOMPurify.sanitize(options.data)) : null;
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js:838]: Are we checking somewhere else that the user does not add script tags and such?

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js (hunk 1 above)
Comment (Jtang-1) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js:13]: The null check is done by [markdown.render](https://github.com/dimagi/commcare-hq/blob/b1a9bef71ab117b24f4166529a255506b0e50b86/corehq/apps/cloudcare/static/cloudcare/js/markdown.js#L152) 
```
md.render(DOMPurify.sanitize(text || "").replaceAll("&#10;", "\n"));
```

**Review by nospame — APPROVED** [review]
Looks clean. Thanks for cleaning up rendering for `description`, too.

**Review by nospame — APPROVED** [review]

**Pull request merged**